### ISSUE TITLE
Facebook Login not working on IOS  Simulator?

### ISSUE THREAD

**Issue by nodesolomon** [invertase/react-native-firebase#53]
Im trying to get the Facebook login to work on IOS simulator (works on a android device :)

the login from Facebook side works as it giving me the access token. I think the problem is with the `firebase.auth().signInWithCredential(credential)` promise is not being resolved.

```
loginFacebook(){
    LoginManager
        .logInWithReadPermissions(['public_profile', 'email'])
        .then((result) => {
          if (result.isCancelled) {
            return Promise.resolve('cancelled');
          }
          console.log(`Login success with permissions: ${result.grantedPermissions.toString()}`);
          // get the access token
          return AccessToken.getCurrentAccessToken();
        })
        .then(data => {
          // create a new firebase credential with the token
          const credential = firebase.auth.FacebookAuthProvider.credential(data.accessToken);

          // login with credential
          return firebase.auth().signInWithCredential(credential);
        })
        .then((currentUser) => { 
         console.log("not being resolved"); // problem is here
          if (currentUser === 'cancelled') {
            console.log('Login cancelled');
          } else {
            // now signed in
            console.warn(JSON.stringify(currentUser.toJSON()));
          }
        })
        .catch((error) => {
          console.log(`Login fail with error: ${error}`);
        });
}
```

Im using react-native 0.43.2.

**Comment by jdmunro** [comment]
I think this line:

```
const credential = firebase.auth.FacebookAuthProvider.credential(data.accessToken);
```

Should be:

```
const credential = firebase.auth().FacebookAuthProvider.credential(data.accessToken);
```

Notice the `auth()` is actually an invocation. Also make sure you've got Facebook authentication enabled in the Firebase Console 😄

**Comment by nodesolomon** [comment]
@jdmunro Thanks for your reply.

I just tried putting invocation on auth:
`const credential = firebase.auth().FacebookAuthProvider.credential(data.accessToken);
`
but Im getting an error saying cannot read property credential of undefined.

<img width="590" alt="screen shot 2017-04-14 at 15 52 45" src="https://cloud.githubusercontent.com/assets/1233465/25046705/da567d90-212a-11e7-8e98-57271af7a2af.png">

I did enable the facebool login on the firebase console, BTW i got this code from the documentation.

thanks again

**Comment by andfk** [comment]
Notice the way the authentication works in `react-native-firebase` is a little different than `firebase`. As you don't need to use `const credential = firebase.auth().FacebookAuthProvider.credential(data.accessToken);`. Instead something like once you have the access token from FB:
```javascript
const credential = {
      provider: 'facebook',
      token: accessToken,
      secret: '<YOUR-FB-APP-SECRET>'
    };
    const user = await firebase.auth().signInWithCredential(credential);
```
You can take a look [here](https://github.com/invertase/react-native-firebase/blob/master/docs/api/authentication.md#signinwithcredentialcredential-object-promise)

**Comment by nodesolomon** [comment]
@andfk thanks

but its not working for me:

LoginManager
        .logInWithReadPermissions(['public_profile', 'email'])
        .then((result) => {
          if (result.isCancelled) {
            return Promise.resolve('cancelled');
          }
          console.log(`Login success with permissions: ${result.grantedPermissions.toString()}`);
          // get the access token
          return AccessToken.getCurrentAccessToken();
        })
        .then(data => {
          // create a new firebase credential with the token

          const credential = {
              provider: 'facebook',
              token: data.accessToken,
              secret: '-----------'
          };

          return firebase.auth().signInWithCredential(credential);
          // login with credential
          
        })
        .then((currentUser) => {
          console.log("not being resolved")
          if (currentUser === 'cancelled') {
            console.log('Login cancelled');
          } else {
            // now signed in
            console.warn(JSON.stringify(currentUser.toJSON()));
          }
        })
        .catch((error) => {
          console.log(`Login fail with error: ${error}`);
        });

**Comment by andfk** [comment]
Can you paste here the error when `firebase.auth().signInWithCredential(credential)` is called? so i can try to help

**Comment by nodesolomon** [comment]
@andfk  its not giving me any errors, thats the problem

```
.then((currentUser) => {
          console.log("not being resolved")
          if (currentUser === 'cancelled') {
            console.log('Login cancelled');
          } else {
            // now signed in
            console.warn(JSON.stringify(currentUser.toJSON()));
          }
        })
```

this piece of code never triggers. Have you tried it on your simulator?

(sorry for the late reply)

**Comment by designorant** [comment]
**TL;DR:** Make sure you're on the latest version, and remove/reinstall your `node_modules`, particularly if you use Yarn.

---

I haven't looked into this issue properly, but I recently stumbled upon something that may be related to this.

I use Facebook auth. It works.

I also use Yarn.

`yarn outdated` shows this:
```
Package                  Current       Wanted       Latest        Package Type    URL
react-native-firebase    1.0.0-alpha13 1.0.0-alpha9 1.0.0-alpha13 dependencies    https://github.com/invertase/react-native-firebase#readme
```

Notice **Wanted** there.

Upgrading often ends up like that in `Yarn.lock`:
```
react-native-firebase@^1.0.0-alpha12:
  version "1.0.0-alpha9"
  resolved "https://registry.yarnpkg.com/react-native-firebase/-/react-native-firebase-1.0.0-alpha9.tgz#9ced13e8f5ea56b4de19611f4d1beabb22729009"
```

For some reason Yarn recognises the latest version (sorted by date, I suppose) but chooses one sorted out alphabetically (?). I don't know whether this is an issue with Yarn itself or our versioning (likely the former), but I've noticed [others](https://www.npmjs.com/package/react-navigation) use versioning like `1.0.0-beta.9` that would likely prevent that.

Whether this is the cause for this issue, I'm not sure but my auth stops working due to a non existing credential every time I'm caught by this, and it's fine as soon as I make sure I'm on the latest version.

Hope this helps.

**Comment by Ehesp** [comment]
Closing due to implementation being incorrect. Please follow designorant's post above, if you still have issues let us know and we can reopen.
